### ISSUE TITLE
[Bug] Triton attention illegal memory access for large token count

### ISSUE THREAD

**Issue by akhilg-nv** [sgl-project/sglang#20389]
### Checklist

- [x] I searched related issues but found no solution.
- [x] The bug persists in the latest version.
- [x] Issues without environment info and a minimal reproducible demo are hard to resolve and may receive no feedback.
- [ ] If this is not a bug report but a general question, please start a discussion at https://github.com/sgl-project/sglang/discussions. Otherwise, it will be closed.
- [x] Please use English. Otherwise, it will be closed.

### Describe the bug

`triton_backend.py` allocates `qo_indptr` as int32. The Triton kernel computes pointer offsets as `qo_indptr[i] * stride_qbs` (where`stride_qbs = num_q_heads * head_dim`). When this product exceeds `2^31-1` (signed int32 max), we consistently encounter a `cudaErrorIllegalAddress` error. It is likely the the result wraps to a negative number, causing the kernel to read from an address that is out of bounds.

Overflow when (total_tokens - 1) * stride > 2^31 - 1, i.e.
total_tokens > 2^31 / stride + 1:
  - Qwen3-Next (stride=8192):  total_tokens > 262,144
  - DeepSeek-V3 (stride=16384): total_tokens > 131,072

### Reproduction

This failure was found through unit kernel benchmarking. Below is a minimal script to reproduce the error.

Minimal repro script:
```
"""
Usage:
    python repro_qo_indptr_overflow.py --model qwen3-next
    python repro_qo_indptr_overflow.py --model qwen3-next --subprocess
    python repro_qo_indptr_overflow.py --model deepseek-v3 --verbose
"""

import argparse
import json
import subprocess
import sys
from pathlib import Path

sys.path.insert(0, str(Path(__file__).resolve().parent.parent))

CONFIGS = {
    "qwen3-next": (32, 2, 256),
    "deepseek-v3": (128, 1, 128),
}

SEQ_LEN = 512

BATCH_SIZES = {
    "qwen3-next": [1, 64, 256, 384, 448, 496, 508, 511, 512,
                   513, 514, 516, 528, 576, 768, 1024],
    "deepseek-v3": [1, 64, 128, 192, 224, 240, 248, 252, 254, 255, 256,
                    257, 258, 260, 264, 272, 288, 320, 384, 512],
}


def run_test(bs, seq_len, num_q, num_kv, dim):
    import torch
    from sglang.srt.layers.attention.triton_ops.extend_attention import extend_attention_fwd

    total = bs * seq_len
    sm_scale = 1.0 / (dim ** 0.5)

    def run_once(qo_dtype):
        torch.manual_seed(42)
        q = torch.randn(total, num_q, dim, dtype=torch.bfloat16, device="cuda")
        k = torch.randn(total, num_kv, dim, dtype=torch.bfloat16, device="cuda")
        v = torch.randn(total, num_kv, dim, dtype=torch.bfloat16, device="cuda")
        o = torch.zeros(total, num_q, dim, dtype=torch.bfloat16, device="cuda")
        qo_indptr = torch.arange(0, bs + 1, dtype=qo_dtype, device="cuda") * seq_len
        kv_indptr = torch.zeros(bs + 1, dtype=torch.int32, device="cuda")
        kv_indices = torch.empty(0, dtype=torch.int64, device="cuda")
        extend_attention_fwd(
            q, k, v, o, k, v, qo_indptr, kv_indptr, kv_indices,
            None, True, None, seq_len,
            k_scale=1.0, v_scale=1.0, sm_scale=sm_scale,
        )
        torch.cuda.synchronize()
        return o

    o_correct = run_once(torch.int64)
    try:
        o_buggy = run_once(torch.int32)
    except Exception as e:
        return "crash", str(e)

    diff = (o_buggy.float() - o_correct.float()).abs()
    max_diff = diff.max().item()
    if max_diff < 0.01:
        return "ok", f"max_diff={max_diff:.6f}"
    pct = (diff > 0.01).float().mean().item() * 100
    return "corruption", f"max_diff={max_diff:.4f} wrong={pct:.1f}%"


def run_in_subprocess(bs, seq_len, num_q, num_kv, dim):
    script = (
        f"import sys, json; sys.path.insert(0, '{Path(__file__).resolve().parent.parent}');"
        f"from repro_qo_indptr_overflow import run_test;"
        f"r, d = run_test({bs}, {seq_len}, {num_q}, {num_kv}, {dim});"
        f"print(json.dumps({{'result': r, 'details': d}}))"
    )
    try:
        proc = subprocess.run(
            [sys.executable, "-c", script], capture_output=True, text=True,
            timeout=120, cwd=str(Path(__file__).resolve().parent),
        )
        if proc.returncode == 0:
            for line in proc.stdout.strip().split("\n"):
                if line.startswith("{"):
                    data = json.loads(line)
                    return data["result"], data["details"]
        stderr = proc.stderr[-300:] if proc.stderr else ""
        if "illegal memory access" in stderr or "cudaError" in stderr:
            return "crash", "CUDA illegal memory access"
        return "crash", f"exit {proc.returncode}"
    except subprocess.TimeoutExpired:
        return "error", "timeout"


def main():
    parser = argparse.ArgumentParser()
    parser.add_argument("--model", required=True, choices=list(CONFIGS.keys()))
    parser.add_argument("--subprocess", action="store_true",
                        help="Run each config in a separate process (slower, survives crashes)")
    parser.add_argument("--verbose", action="store_true",
                        help="Print error details for failures")
    args = parser.parse_args()

    num_q, num_kv, dim = CONFIGS[args.model]
    stride = num_q * dim
    int32_max = 2**31 - 1

    print(f"{args.model}: {num_q} Q heads, {num_kv} KV heads, dim={dim}, stride={stride}")
    max_safe_tokens = int32_max // stride + 1
    print(f"Overflow when total_tokens > {max_safe_tokens:,} (i.e. (tok-1)*stride > 2^31-1)\n")

    print(f"{'bs':>6}  {'seq':>6}  {'total_tok':>12}  {'(tok-1)*stride':>20}  {'vs 2^31-1':>16}  result")
    print("-" * 85)

    for bs in BATCH_SIZES[args.model]:
        total = bs * SEQ_LEN
        max_idx = total - 1
        product = max_idx * stride
        delta = product - int32_max

        if args.subprocess:
            result, details = run_in_subprocess(bs, SEQ_LEN, num_q, num_kv, dim)
        else:
            result, details = run_test(bs, SEQ_LEN, num_q, num_kv, dim)

        marker = "" if result == "ok" else f"  <-- {result.upper()}"
        delta_str = f"{delta:>+16,}"
        print(f"{bs:>6}  {SEQ_LEN:>6}  {total:>12,}  {product:>20,}  {delta_str}  {result}{marker}")

        if args.verbose and result != "ok":
            print(f"           {details}")

        if result == "crash" and not args.subprocess:
            print("\n  (GPU context poisoned — use --subprocess to continue past crashes)")
            break


if __name__ == "__main__":
    main()
```

SGLang's scheduler caps total extend tokens per batch via `chunked_prefill_size`. Even with aggressive tuning, activation memory limits and sequential prefill processing prevent batches from reaching the threshold. However, the fix is a one-line int32→int64 (change `qo_indptr` to `int64`) with zero performance cost, and models with larger stride (e.g. future MLA variants) would lower the threshold.

### Environment

8xh200 in sglang nightly dev container from today
```
Python: 3.12.3 (main, Mar  3 2026, 12:15:18) [GCC 13.3.0]                                                                         
CUDA available: True                                                                                                              
GPU 0,1,2,3,4,5,6,7: NVIDIA H200                                                                                                  
GPU 0,1,2,3,4,5,6,7 Compute Capability: 9.0                                                                                       
CUDA_HOME: /usr/local/cuda                                                                                                        
NVCC: Cuda compilation tools, release 12.9, V12.9.86                                                                              
CUDA Driver Version: 590.48.01                                                                                                    
PyTorch: 2.9.1+cu129                                                                                                              
sglang: 0.0.0.dev1+g7a1ca5380                                                                                                     
sgl_kernel: 0.3.21                                                                                                                
flashinfer_python: 0.6.4                                                                                                          
flashinfer_cubin: 0.6.4                                                                                                           
flashinfer_jit_cache: 0.6.4+cu129                                                                                                 
triton: 3.5.1                                                                                                                     
transformers: 4.57.1                                                                                                              
torchao: 0.9.0                                                                                                                    
numpy: 2.4.3                                                                                                                      
aiohttp: 3.13.3                                                                                                                   
fastapi: 0.135.1                                                                                                                  
hf_transfer: 0.1.9                                                                                                                
huggingface_hub: 0.36.2                                                                                                           
interegular: 0.3.3                                                                                                                
modelscope: 1.34.0
orjson: 3.11.7
outlines: 0.1.11
packaging: 26.0
psutil: 7.2.2
pydantic: 2.12.5
python-multipart: 0.0.22
pyzmq: 27.1.0
uvicorn: 0.41.0
uvloop: 0.22.1
vllm: Module Not Found
xgrammar: 0.1.27
openai: 2.6.1
tiktoken: 0.12.0
anthropic: 0.84.0
litellm: Module Not Found
torchcodec: 0.9.1
NVIDIA Topology: 
        GPU0    GPU1    GPU2    GPU3    GPU4    GPU5    GPU6    GPU7    NIC0    NIC1    NIC2    NIC3    NIC4    NIC5    NIC6    NI
C7      NIC8    NIC9    NIC10   NIC11   CPU Affinity    NUMA Affinity   GPU NUMA ID
GPU0     X      NV18    NV18    NV18    NV18    NV18    NV18    NV18    PXB     NODE    NODE    NODE    NODE    NODE    SYS     SY
S       SYS     SYS     SYS     SYS     0-55,112-167    0               N/A
GPU1    NV18     X      NV18    NV18    NV18    NV18    NV18    NV18    NODE    NODE    NODE    PXB     NODE    NODE    SYS     SY
S       SYS     SYS     SYS     SYS     0-55,112-167    0               N/A
GPU2    NV18    NV18     X      NV18    NV18    NV18    NV18    NV18    NODE    NODE    NODE    NODE    PXB     NODE    SYS     SY
S       SYS     SYS     SYS     SYS     0-55,112-167    0               N/A
GPU3    NV18    NV18    NV18     X      NV18    NV18    NV18    NV18    NODE    NODE    NODE    NODE    NODE    PXB     SYS     SY
S       SYS     SYS     SYS     SYS     0-55,112-167    0               N/A
GPU4    NV18    NV18    NV18    NV18     X      NV18    NV18    NV18    SYS     SYS     SYS     SYS     SYS     SYS     PXB     NO
DE      NODE    NODE    NODE    NODE    56-111,168-223  1               N/A
GPU5    NV18    NV18    NV18    NV18    NV18     X      NV18    NV18    SYS     SYS     SYS     SYS     SYS     SYS     NODE    NO
DE      NODE    PXB     NODE    NODE    56-111,168-223  1               N/A
GPU6    NV18    NV18    NV18    NV18    NV18    NV18     X      NV18    SYS     SYS     SYS     SYS     SYS     SYS     NODE    NO
DE      NODE    NODE    PXB     NODE    56-111,168-223  1               N/A
GPU7    NV18    NV18    NV18    NV18    NV18    NV18    NV18     X      SYS     SYS     SYS     SYS     SYS     SYS     NODE    NO
DE      NODE    NODE    NODE    PXB     56-111,168-223  1               N/A
NIC0    PXB     NODE    NODE    NODE    SYS     SYS     SYS     SYS      X      NODE    NODE    NODE    NODE    NODE    SYS     SY
S       SYS     SYS     SYS     SYS
NIC1    NODE    NODE    NODE    NODE    SYS     SYS     SYS     SYS     NODE     X      PIX     NODE    NODE    NODE    SYS     SY
S       SYS     SYS     SYS     SYS
NIC2    NODE    NODE    NODE    NODE    SYS     SYS     SYS     SYS     NODE    PIX      X      NODE    NODE    NODE    SYS     SY
S       SYS     SYS     SYS     SYS
NIC3    NODE    PXB     NODE    NODE    SYS     SYS     SYS     SYS     NODE    NODE    NODE     X      NODE    NODE    SYS     SY
S       SYS     SYS     SYS     SYS
NIC4    NODE    NODE    PXB     NODE    SYS     SYS     SYS     SYS     NODE    NODE    NODE    NODE     X      NODE    SYS     SY
S       SYS     SYS     SYS     SYS
NIC5    NODE    NODE    NODE    PXB     SYS     SYS     SYS     SYS     NODE    NODE    NODE    NODE    NODE     X      SYS     SY
S       SYS     SYS     SYS     SYS
NIC6    SYS     SYS     SYS     SYS     PXB     NODE    NODE    NODE    SYS     SYS     SYS     SYS     SYS     SYS      X      NO
DE      NODE    NODE    NODE    NODE
NIC7    SYS     SYS     SYS     SYS     NODE    NODE    NODE    NODE    SYS     SYS     SYS     SYS     SYS     SYS     NODE     X
        PIX     NODE    NODE    NODE
NIC8    SYS     SYS     SYS     SYS     NODE    NODE    NODE    NODE    SYS     SYS     SYS     SYS     SYS     SYS     NODE    PI
X        X      NODE    NODE    NODE
NIC9    SYS     SYS     SYS     SYS     NODE    PXB     NODE    NODE    SYS     SYS     SYS     SYS     SYS     SYS     NODE    NO
DE      NODE     X      NODE    NODE
NIC10   SYS     SYS     SYS     SYS     NODE    NODE    PXB     NODE    SYS     SYS     SYS     SYS     SYS     SYS     NODE    NO
DE      NODE    NODE     X      NODE
NIC11   SYS     SYS     SYS     SYS     NODE    NODE    NODE    PXB     SYS     SYS     SYS     SYS     SYS     SYS     NODE    NO
DE      NODE    NODE    NODE     X 

Legend:

  X    = Self
  SYS  = Connection traversing PCIe as well as the SMP interconnect between NUMA nodes (e.g., QPI/UPI)
  NODE = Connection traversing PCIe as well as the interconnect between PCIe Host Bridges within a NUMA node
  PHB  = Connection traversing PCIe as well as a PCIe Host Bridge (typically the CPU)
  PXB  = Connection traversing multiple PCIe bridges (without traversing the PCIe Host Bridge)
  PIX  = Connection traversing at most a single PCIe bridge
  NV#  = Connection traversing a bonded set of # NVLinks

NIC Legend:

  NIC0: mlx5_0
  NIC1: mlx5_1
  NIC2: mlx5_2
  NIC3: mlx5_3
  NIC4: mlx5_4
  NIC5: mlx5_5
  NIC6: mlx5_6
  NIC7: mlx5_7
  NIC8: mlx5_8
  NIC9: mlx5_9
  NIC10: mlx5_10
  NIC11: mlx5_11


ulimit soft: 1048576
```
